File: Assignment3/Assignment3_3.c
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name: Assignment3_3.c 
// Description: Write a program which accept number from user and display all even factors of that number. 
// Author: Aniket Devaba Markad 
// Date: 11/05/2025
// 
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 

#include<stdio.h>

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 
// 
// Function Name : DisplayEvenFactors 
// Description   : This function is used to display even  factors of number. 
// Input         : int 
// Output        : int 
// Author        : Aniket Devaba Markad 
// Date          : 11/05/2025 
// 
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

void DisplayEvenFactor(int iNo)

{
    int iCnt = 0;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    for(iCnt = 1; iCnt < iNo ;iCnt++)
    {
        if(((iNo % iCnt) == 0) && ((iCnt % 2) == 0))
        {
            printf("%d\n",iCnt);
        }
    }
}
int main()
{
    int iValue = 0;

    printf("Enter Number:");
    scanf("%d",&iValue);

    DisplayEvenFactor(iValue);

    return 0;
}

File: Java/Assignment3/Assignment3_3.java

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// file name   : Assignment3_3.java
// Description : write a program which accept one number from user and print  odd factors of that numbers on screen.
// Author      : Aniket Devaba Markad                        
// Date        : 28/11/2025 
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.*;

class number
{

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name : DisplayOddFactor 
// Description   : This function is used to print even factor of number numbers. 
// Author        : Aniket Devaba Markad 
// Date          : 28/11/2025 
// 
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

    void DisplayOddFactor(int iNo)
    {
        int iCnt = 0;

        if(iNo < 0)
        {
            iNo = -(iNo);     //updator
        }
        System.out.println("Odd factor of "+iNo+" : " );

        for(iCnt = 1;iCnt <= iNo/2; iCnt++)
        {
            if(iNo % iCnt == 0)
            {
                if(iCnt % 2 != 0)
                {
                    System.out.println(iCnt);
                }
            }
        }
    }
}

public class Assignment3_3
{
    public static void main(String a[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the number");
        iValue = sobj.nextInt();

        number nobj = new number();
        nobj.DisplayOddFactor(iValue);
    }
}




